values.yaml

```
@@ -655,39 +670,39 @@ ruler:
   # -- allow configuring rules via configmap. ref: https://cortexproject.github.io/cortex-helm-chart/guides/configure_rules_via_configmap.html
   directories: {}
 
-  # -- Sidecars that collect the configmaps with specified label and stores the included files them into the respective folders
   sidecar:
+    # -- Enable sidecar that collect the configmaps with specified label and stores the included files them into the respective folders
+    enabled: false
     image:
-      repository: quay.io/kiwigrid/k8s-sidecar
-      tag: 1.10.7
+      repository: omegavveapon/kopf-k8s-sidecar
+      tag: 1.4.0
       sha: ""
     imagePullPolicy: IfNotPresent
     resources: {}
-    #   limits:
-    #     cpu: 100m
-    #     memory: 100Mi
-    #   requests:
-    #     cpu: 50m
-    #     memory: 50Mi
-    # skipTlsVerify Set to true to skip tls verification for kube api calls
-    # skipTlsVerify: true
+    # -- Set to true to skip tls verification for kube api calls
+    skipTlsVerify: false
+    # -- A value of true will produce unique filenames to avoid issues when duplicate data keys exist between ConfigMaps
+    # and/or Secrets within the same or multiple Namespaces.
     enableUniqueFilenames: false
-    enabled: false
     # -- label that the configmaps with rules are marked with
     label: cortex_rules
+    # -- Determines how kopf-k8s-sidecar will run. If WATCH it will run like like a normal operator forever.
+    # If LIST it will gather the matching configmaps and secrets currently present,
+    # write those files to the destination directory and die
     watchMethod: null
-    # -- value of label that the configmaps with rules are set to
+    # -- The value for the label you want to filter your resources on.
+    # Don't set a value to filter by any value
     labelValue: null
-    # -- folder in the pod that should hold the collected rules (unless `defaultFolderName` is set)
+    # -- Folder where the files should be placed.
     folder: /tmp/rules
     # -- The default folder name, it will create a subfolder under the `folder` and put rules in there instead
     defaultFolderName: null
-    # -- If specified, the sidecar will search for rules config-maps inside this namespace.
-    # Otherwise the namespace in which the sidecar is running will be used.
-    # It's also possible to specify ALL to search in all namespaces
+    # -- The Namespace(s) from which resources will be watched.
+    # For multiple namespaces, use a comma-separated string like "default,test".
+    # If not set or set to ALL, it will watch all Namespaces.
```

```suggestion
    # If not set or set to ALL, it will watch all Namespaces.
```
that's different than before.

---

Yeah, that's what I was saying in my comment from Friday. This is the only gotcha in this alternative sidecar. Strangely, the author decided to break compatibility on this one parameter 🤷 . It's the only sidecar parameter that we currently override from the current defaults here in the Cortex Helm chart. We did not want the new default behavior to watch all namespaces. So if you wanted to make this a totally backwards compatible change, then you would need to set it explicitly to `{{ .Release.Namespace }}`. Not sure it's worth that effort though. It's a pretty easy release note for an existing user to adjust to.

---

Yeah I don't see it as a big problem either. The guide has it as an example and its well documented